IdentityCore/src/network/session_delegate/MSIDURLSessionDelegate.m

```diff
@@ -25,4 +25,74 @@
 
 @implementation MSIDURLSessionDelegate
 
+#pragma mark - NSURLSessionDelegate
+
+- (void)URLSession:(NSURLSession *)session didReceiveChallenge:(NSURLAuthenticationChallenge *)challenge
+ completionHandler:(void (^)(NSURLSessionAuthChallengeDisposition disposition, NSURLCredential * _Nullable credential))completionHandler
+{
+    NSString *authMethod = [challenge.protectionSpace.authenticationMethod lowercaseString];
+    
+    MSID_LOG_VERBOSE(nil, @"%@ - %@. Previous challenge failure count: %ld", @"session:didReceiveChallenge:completionHandler", authMethod, (long)challenge.previousFailureCount);
+    
+    NSURLSessionAuthChallengeDisposition disposition = NSURLSessionAuthChallengePerformDefaultHandling;
+    __block NSURLCredential *credential = nil;
+    
+    if (self.sessionDidReceiveAuthenticationChallengeBlock)
+    {
+        disposition = self.sessionDidReceiveAuthenticationChallengeBlock(session, challenge, &credential);
```

Comment:
nit: This is a bit different from how interactive challenge handlers are implemented, where challenge handler calls the completionHandler itself. Can we make sure they both behave the same? I have no personal preference over one or another.